### PR TITLE
Add class file to get source packages.

### PR DESCRIPTION
**Pull request by Yoshihiro-Okada-ML**
# Purpose of pull request

This class downloads source code and debian related files for all packages which are included in rootfs.

# Test
## How to test

Add following line into the local.conf.

```
INHERIT += "dl-srcpkg"
```

And then type following build command.

```
$ bitbake emlinux-image-base
```

## Test result

All source code and debian related packages are located under the following directory.

```
tmp/deploy/sources/xxxxx
```

### CODE REVIEW

File: classes/dl-srcpkg.bbclass [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    ### prepare directory for output
+    mkdir -p ${ROOTFS_MANIFEST_DEPLOY_DIR}/sources
+    sudo -E mount --bind /tmp '${ROOTFSDIR}'/tmp
+    mkdir -p '${ROOTFSDIR}'/tmp/deploy_dir
+    sudo -E mount --bind '${ROOTFS_MANIFEST_DEPLOY_DIR}' '${ROOTFSDIR}'/tmp/deploy_dir
+
+    ### setup network & apt environment
+    sudo cp /etc/resolv.conf '${ROOTFSDIR}'/etc/resolv.conf
+    prepare_local_isarapt
+    sudo sh -c "echo 'deb [trusted=yes] copy:///tmp/isar-apt/emlinux-bookworm-arm64/apt/emlinux-bookworm isar main' > '${ROOTFSDIR}'/etc/apt/sources.list.d/isar-apt.list"
```
Comment (masami256) [inline classes/dl-srcpkg.bbclass]: please fix hardcoded architecture name in line 38 and 39.

Comment (Yoshihiro-Okada-ML) [reply classes/dl-srcpkg.bbclass]: I replaced some hardcoded strings and omitted some needless quote.

File: classes/dl-srcpkg.bbclass [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+# SPDX-License-Identifier: MIT
+#
+
+#
+# [How to use]
+# Add following line into the local.conf.
+# INHERIT += "dl-srcpkg"
+#
+
+LOCAL_APT_DIR="tmp/isar-apt/${DISTRO}-${DISTRO_ARCH}/apt/${DISTRO}"
```
Comment (masami256) [inline classes/dl-srcpkg.bbclass]: tmp directory is changeable so ${TMPDIR} would be better.

**Comment by masami256** [comment]
> All source code and debian related packages are located under the following directory.
> tmp/deploy/images/xxxxx/sources

How about put files one of following directory instead of tmp/deploy/images/<MACHINE>/sources
- tmp/deploy/sources
- tmp/deploy/<MACHINE>/sources
- tmp/deploy/sources/<MACHINE>

File: classes/dl-srcpkg.bbclass (hunk 2 above)
Comment (Yoshihiro-Okada-ML) [reply classes/dl-srcpkg.bbclass]: TMPDIR is a absolute path from "/" on HOST PC environment. This LOCAL_APT_DIR is relative directory in rootfs environment. So we can't use TMPDIR.

File: classes/dl-srcpkg.bbclass (hunk 1 above)
Comment (masami256) [reply classes/dl-srcpkg.bbclass]: thanks.

**Comment by masami256** [comment]
LGTM

**Comment by hiraku-wfs** [comment]
LGTM

**Comment by masami256** [comment]
@Yoshihiro-Okada-ML thanks.

**Pull request merged**